handlers/hub_test.go

```diff
@@ -0,0 +1,341 @@
+package handlers_test
+
+import (
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"path"
+	"regexp"
+	"testing"
+	"time"
+
+	"github.com/jmoiron/sqlx"
+	"github.com/ripple-cloud/cloud/data"
+	"github.com/ripple-cloud/cloud/handlers"
+	"github.com/ripple-cloud/cloud/router"
+	"github.com/ripple-cloud/cloud/testhelpers"
+)
+
+func setupServerAddHub(db *sqlx.DB, tokenSecret []byte) (*httptest.Server, error) {
+	r := router.New()
+
+	r.Default(
+		handlers.SetConfig(db, []byte(tokenSecret)),
+	)
+
+	r.GET("/api/v0/hub", handlers.Auth, handlers.AddHub)
+
+	return httptest.NewServer(r), nil
+}
+
+func setupServerShowHub(db *sqlx.DB, tokenSecret []byte) (*httptest.Server, error) {
+	r := router.New()
+
+	r.Default(
+		handlers.SetConfig(db, []byte(tokenSecret)),
+	)
+
+	r.GET("/api/v0/hub", handlers.Auth, handlers.ShowHub)
+
+	return httptest.NewServer(r), nil
+}
+
+func setupServerDeleteHub(db *sqlx.DB, tokenSecret []byte) (*httptest.Server, error) {
+	r := router.New()
+
+	r.Default(
+		handlers.SetConfig(db, []byte(tokenSecret)),
+	)
+
+	r.GET("/api/v0/hub", handlers.Auth, handlers.DeleteHub)
+
+	return httptest.NewServer(r), nil
+}
+
+func TestAddHub(t *testing.T) {
+	// setup DB
+	db := testhelpers.SetupDB(t)
+	defer db.Close()
+
+	// setup server
+	ts, err := setupServerAddHub(db, []byte("secret"))
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer ts.Close()
+
+	// create a user
+	u := &data.User{
+		Username: "foo",
+		Email:    "foo@example.com",
+	}
+	if err := u.EncryptPassword("password"); err != nil {
+		t.Fatal(err)
+	}
+	if err = u.Insert(db); err != nil {
+		t.Fatal(err)
+	}
+
+	// create a token for the user
+	tok := data.Token{
+		UserID:    u.ID,
+		ExpiresIn: (30 * 24 * time.Hour).Nanoseconds(), // 30 days
+	}
+	if err := tok.Insert(db); err != nil {
+		t.Fatal(err)
+	}
+
+	// get the encoded JSON Web Token
+	jwt, err := tok.EncodeJWT([]byte("secret"))
+	if err != nil {
+		t.Fatal(err)
+	}
```

Comment:
Guess creating a user & adding token would be repeated quite often. Better we move them to a test helper. (not necessarily in this PR).